diesel/src/sqlite/connection/raw.rs

```diff
@@ -163,6 +163,22 @@ impl RawConnection {
             ))
         }
     }
+
+    pub fn enable_load_extension(&self, enabled: bool) -> QueryResult<()> {
+        let result = unsafe {
+            ffi::sqlite3_enable_load_extension(self.internal_connection.as_ptr(), enabled as i32)
```

Comment:
@diesel-rs/reviewers Seems like this function is not available on all supported platforms. For example it is missing in the default installation on MacOS and there is a configure flag to disable this function. Now this opens the question how we should handle this. I see the following options:

* Don't implement this feature, which the obvious downsides of not supporting extension loading on sqlite.
* Drop support for sqlite3 versions that do not have this functions. Not sure how large the impact of that would be, as you can just bundle sqlite3 in your program.
* Do some checking here if that function is available.
    + On compile time, as this is a property of the provided library, but how would one implement that in a portable way.
    + As sqlite3 could be dynamically linked compile time checking could break, so we would need to do a dynamic lookup here, but against which library and how would that work with static linking

Reply:
@weiznich , you've wrote in https://github.com/diesel-rs/diesel/discussions/3925#discussioncomment-8386629
> For extensions there is #2180 which basically stalled because I found no way to detect at build time if the underlying sqlite library was build with extension support enabled or disabled.

Is it is not enough just to load sqlite library and try to get desired function using `GetProcAddress` / `dlsym`? Or maybe just try to compile a small program that links to that library and depends on the presence of that function and check the compiler status code?

Reply:
I remember playing around with these functions back then while trying to get this working. It helps for builds that dynamically link libsqlite3, but it is problematic for static builds. Also it did not work for musl based builds. At that point I stopped trying to do that.

As for the test build option: That actually might work.

Reply:
In `bundled` mode you've already known which sqlite features are enabled or no? If it is compiled during build, maybe you even can control which build flags to use

Reply:
That might work with the `bundled` feature, if it's even possible to detect that it's set on a dependency. (We could likely patch `libsqlite3-sys` to expose that). It won't work if the user supplies another statically build `libsqlite3.a` file on it's own. It also won't work for cross compilation.